src/attach.rs

```diff
@@ -85,25 +88,50 @@ impl Attached {
         Db: ?Sized + Database,
     {
         struct DbGuard<'s> {
-            state: &'s Attached,
+            state: Option<&'s Attached>,
```

Comment:
What's the reason to make this optional? It's not evident to me how it's related to the change (but it probably it is, just not evident to me)

Reply:
I would find it useful if we document the different states that are possible (what does `state: Some`, `prev: None` mean etc)